
syntax = "proto3";


message BaseRequest {
	int32 version = 1;
	int32 length = 2;
	bytes buffer = 3;
}

message BaseReply {
	int32 version = 1;
	int32 length = 2;
	bytes buffer = 3;
}


service Base 
{
	rpc BaseTransmit (BaseRequest) returns (BaseReply) {}

	rpc ServerStreamTransmit (BaseRequest) returns (stream BaseReply) {}

	rpc ClientStreamTransmit(stream BaseRequest) returns (BaseReply) {}

	rpc StreamTransmit (stream BaseRequest) returns (stream BaseReply) {}
}
